### PR TITLE
multi: update to latest deps

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: backend/stakepoold/rpcclient.go [hunk 1]
```diff
@@ -134,30 +136,19 @@ func walletGetTickets(spd *stakepool.Stakepoold) (map[chainhash.Hash]string, map
...
-		rpcclient.FutureGetTransactionResult
-	}
-	promises := make([]promise, 0, len(tickets))
-
-	log.Debugf("setting up GetTransactionAsync for %v tickets", len(tickets))
-	for _, ticket := range tickets {
-		// lookup ownership of each ticket
-		promises = append(promises, promise{spd.WalletConnection.RPCClient().GetTransactionAsync(ticket)})
-	}
-
```
Comment (jrick) [inline backend/stakepoold/rpcclient.go:155]: the previous code started all rpc calls without waiting for any of them to complete, and then waited on each of those calls finishing before processing the result.  this updated version will serialize all of the calls which would dramatically slow down this procedure.  in order to regain the async nature of this code, each call should be executed as its own goroutine, saving the results to a slice (probably) where each index of that slice corresponds to a started call.  You can use a errgroup or waitgroup to wait until all of the calls have finished (either in success or error) before iterating through all of the errors and returned results below.

File: backend/stakepoold/stakepool/client.go [hunk 2]
```diff
@@ -5,7 +5,9 @@ import (
 	"sync"
 	"time"
 
-	"github.com/decred/dcrd/rpcclient/v5"
+	walletRPC "decred.org/dcrwallet/rpc/client/dcrwallet"
```
Comment (jrick) [inline backend/stakepoold/stakepool/client.go:8]: I have never seen a package name use capital letters so this seems a bit not idomatic.

Unless there is a technical limitation to overcome, I'd recommend using the real package name here (`dcrwallet`).

**Comment by jholdstock** [comment]
Which dcrwallet version is this supposed to work with? Tried release 1.5.1 (RPC version 6.2.0) and master (RPC version 8.0.0) but this PR seems to be looking for version 7.0.0

File: backend/stakepoold/stakepool/client.go [hunk 3]
```diff
@@ -66,15 +70,16 @@ func (c *Client) IsConnected() bool {
 
 // RPCClient allows access to the underlying rpcclient by providing a copy of
 // its address.
-func (c *Client) RPCClient() *rpcclient.Client {
+func (c *Client) RPCClient() *dcrwallet.Client {
 	c.mux.RLock()
 	defer c.mux.RUnlock()
-	return c.client
+	raw := dcrwallet.RawRequestCaller(c.client)
```
[the remark this answers is not in the document]
Comment (dajohi) [reply backend/stakepoold/stakepool/client.go:76]: dcrwallet.Client is just a wrapper for RPC calls; it doesn't handle connect/disconnect.

i have another branch that converts dcrstakepool to replace github.com/decred/dcrd/rpcclient with github.com/jrick/wsrpc.

File: backend/stakepoold/rpcclient.go [hunk 4]
```diff
@@ -134,33 +137,49 @@ func walletGetTickets(spd *stakepool.Stakepoold) (map[chainhash.Hash]string, map
...
 	log.Debugf("setting up GetTransactionAsync for %v tickets", len(tickets))
-	for _, ticket := range tickets {
-		// lookup ownership of each ticket
-		promises = append(promises, promise{spd.WalletConnection.RPCClient().GetTransactionAsync(ticket)})
-	}
+
+	txChan := make(chan *wallettypes.GetTransactionResult, len(tickets))
+	go func() {
+		numGC := 32
```
[the remark this answers is not in the document]
Comment (dajohi) [reply backend/stakepoold/rpcclient.go:152]: Before, the code would call GetTransactionAsync on all tickets.  rpcclient would buffer these calls to 50.  https://github.com/decred/dcrd/blob/master/rpcclient/infrastructure.go#L81

The new code calls GetTransaction on `numGC` txs are a time and waits for those to complete before the next iteration.  As the calls complete, they are processed in the loop below `gt := <-txChan`

**Comment by dajohi** [comment]
bumped

File: backend/stakepoold/stakepool/client.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply backend/stakepoold/stakepool/client.go:76]: me too...

**Pull request merged**